### DISCUSSION TITLE
求大佬科普，xudp到底是个什么东西

### DISCUSSION THREAD

**Discussion by shandongtlb** [XTLS/Xray-core#2687] [Q&A] [answered]
之前在用singbox，其中客户端有个包编码的选项，开启xudp之后可以大大改善连接质量，想请教一下各位大佬，这个xudp是做什么的，原理是什么呢？

最近想换成xray，可是我发现v2rayng客户端没有开启xudp的选项，感觉延迟普遍比singbox高一些，请问如何在xray开启xudp呢，另外，开启之后会有什么副作用呢？

补充一下，singbox和xray配置均是vless vision utls reality

**Comment by chika0801** [comment, marked as answer]
XUDP帖子 https://github.com/XTLS/Xray-core/discussions/252

sing-box里有选项是因为它要兼容v2fly-core和xray-core 2个core的方式，所以做了区别。

> 最近想换成xray，可是我发现v2rayng客户端没有开启xudp的选项，感觉延迟普遍比singbox高一些，请问如何在xray开启xudp呢，另外，开启之后会有什么副作用呢？

xray配置文件不像sing-box配置文件，没有专门的参数。你两端core都为xray即默认会使用相关特性。

**Comment by us254** [comment]
XUDP is an extension of the Mux.Cool protocol. It was created to bring UDP port and address information into the protocol, thereby facilitating communication. It is named XUDP because it implements FullCone for Mux, which also achieves the original goal of XUDP: to provide a UDP aggregation tunnel. This implementation serves a dual purpose by both extending the Mux.Cool protocol to support UDP traffic and by fulfilling the aim of UDP aggregation. The implementation of XUDP is integrated into the Xray-core and is designed to be a simple change to the frame/session writer/reader client/server to support XUDP.
